pilot/model/conversion.go

```diff
@@ -93,7 +94,8 @@ func (ps *ProtoSchema) FromJSON(js string) (proto.Message, error) {
 
 // ApplyJSON unmarshals a JSON string into a proto message
 func ApplyJSON(js string, pb proto.Message) error {
-	return jsonpb.UnmarshalString(js, pb)
+	m := jsonpb.Unmarshaler{AllowUnknownFields: true}
```

Comment:
Is this code used to handle user specified inputs ?
If so this has the effect of not knowing what went wrong for mistyped keys.
This should print warnings.
Since jsonpb does not support this directly it can be achieved by strict parsing first and logging warning for `UnknownFields`, followed by `AllowUnknownFields`

Reply:
+1 for this. It's too easy to make a typo in a YAML.

Reply:
This is some of the reason I'd like us to be able to build out better tooling for interacting with config. We should be strict at validation time and lenient at runtime, especially for cases like forward upgrades. Ideally an older `istioctl` would've rejected the config (or warned about it), but a running Pilot service would've accepted it. (I think this is one of the big ways we can leverage `istioctl` to provide value over just interacting with Istio config via `kubectl`, btw.)

Reply:
The mesh config is in unfortunate state of not being an istio config. It's a config map.
For CRDs, we have versioned the resources and can be strict about extra fields at both runtime and validation time. We can promote older resources to newer version by filling out default values using k8s machinery (once it becomes available).

Reply:
We plan to move mesh config out of a configmap and into a resource in its own right in the near future though. Even still, istioctl could be used to push the CM into k8s, and provide more meaningful validation on the client side as a result.

Either way for this specific PR we should just use Mandar's suggestion: try strictly first and log warnings, then finally ingest leniently.